source/MaterialXRender/Handlers/GeometryHandler.cpp

```diff
@@ -1,30 +1,62 @@
+#include <MaterialXGenShader/Util.h>
 #include <MaterialXRender/Handlers/GeometryHandler.h>
 
 namespace MaterialX
 { 
-const std::string GeometryHandler::UNIT_QUAD("unit_quad");
-const std::string GeometryHandler::POSITION_ATTRIBUTE("i_position");
-const std::string GeometryHandler::NORMAL_ATTRIBUTE("i_normal");
-const std::string GeometryHandler::TEXCOORD_ATTRIBUTE("i_texcoord");
-const std::string GeometryHandler::TANGENT_ATTRIBUTE("i_tangent");
-const std::string GeometryHandler::BITANGENT_ATTRIBUTE("i_bitangent");
-const std::string GeometryHandler::COLOR_ATTRIBUTE("i_color");
-
-GeometryHandler::GeometryHandler() :
-    _identifier(UNIT_QUAD)
+void GeometryHandler::addLoader(GeometryLoaderPtr loader)
 {
+    const StringVec& extensions = loader->supportedExtensions();
+    for (auto extension : extensions)
+    {
+        _geometryLoaders.insert(std::pair<std::string, GeometryLoaderPtr>(extension, loader));
+    }
 }
 
-GeometryHandler::~GeometryHandler() 
+bool GeometryHandler::loadGeometry(const std::string& fileName)
 {
-}
+    // Early return if file already loaded
+    if (_fileName == fileName)
+        return true;
 
-void GeometryHandler::setIdentifier(const std::string& identifier)
-{
-    if (identifier != _identifier)
+    const float MAX_FLOAT = std::numeric_limits<float>::max();
+
+    // Remove any existing meshes, and reset cached file name
+    _meshes.clear();
```

Comment:
If multiple files are supported, as mentioned above, you could move clearing the handler to a separate clear method.